sdk/iot/hub/inc/az_iot_hub_client.h

```diff
@@ -482,15 +505,19 @@ AZ_NODISCARD az_result az_iot_hub_client_twin_parse_received_topic(
  *
  * @param[in] client The #az_iot_hub_client to use for this call.
  * @param[in] request_id The request id.
- * @param[in] mqtt_topic An empty #az_span with sufficient capacity to hold the MQTT topic.
- * @param[out] out_mqtt_topic The output #az_span containing the MQTT topic.
+ * @param[out] mqtt_topic A char buffer with sufficient capacity to hold the MQTT topic
+ *                              filter. On success, will be `NULL` terminated.
+ * @param[in] mqtt_topic_size The size of the passed buffer. Must be greater than 0.
+ * @param[out] out_mqtt_topic_length The optional output length of the mqtt topic filter. Can
+ * be `NULL`.
  * @return #az_result
  */
-AZ_NODISCARD az_result az_iot_hub_client_twin_get_publish_topic_get(
+AZ_NODISCARD az_result az_iot_hub_client_twin_get_get_publish_topic(

```

Comment:
We need a better name for `twin_get`: `twin_download_document` maybe?

Reply:
updated